src/lf_apply.rs

```diff
@@ -341,6 +341,23 @@ pub(crate) unsafe fn rav1d_copy_lpf<BD: BitDepth>(
     }
 }
 
+/// Slice `[u8; 4]`s from `lvl`, but "unaligned",
+/// meaning the `[u8; 4]`s can straddle
+/// adjacent `[u8; 4]`s in the `lvl` slice.
+///
+/// Note that this does not result in actual unaligned reads,
+/// since `[u8; 4]` has an alignment of 1.
+/// This optimizes to a single slice with a bounds check.
+#[inline(always)]
+fn unaligned_lvl_slice(lvl: &[[u8; 4]], y: usize) -> &[[u8; 4]] {
+    // Safety: `[u8; 4]` is transmutable to `u8`s.
+    let (_, lvl, _) = unsafe { lvl.align_to::<u8>() };
+    let lvl = &lvl[y..];
+    // Safety: `[u8; 4]` is transmutable to `u8`s.
+    let (_, lvl, _) = unsafe { lvl.align_to::<[u8; 4]>() };
+    lvl
+}
+
```

Comment:
Should that be `as_chunks<4>().0` instead of `first_chunk<4>().unwrap()`? Several `[u8;4]` elements will be read by `lf.loop_filter_sb`

Reply:
If `lf.loop_filter_sb` reads multiple elements then even that is not quite right, you'd need to check that there are as many `[u8; 4]` elements in the slice as `lf.loop_filter_sb` will access, right?

Reply:
Yes, was trying to figure out what that number of elements is...

Reply:
Thanks for finding these better methods!  I had known about `.as_chunks()`, and it'd also be very useful other places, so it might be worth it to copy the unstable impl into our code.  I didn't know about `.flatten()`, though; that's super useful as well.

Reply:
I copied those unstable methods from `std` in 180f07b.  We've now removed all unstable features, so I didn't want to add new ones.  Once they're stabilized (hopefully soonish), we can switch to them.